Linkta-org/core
# add firebase auth; add google auth; add RQ mutations; fix husky; add …

## Description
This PR was starting to touch a lot of files and I think this is a good point to commit some changes. 

1.  This PR sets up firebase and Google Auth in the client application. The Google Auth buttons on the Signup and Signin page now work. After being authenticated, user should be navigated to the UserInput Form, a protected Route. 

2. Pre-commit hook Fix.   npm command. After diving deeper into  [this ](https://github.com/typicode/husky/issues/1016)issue, we are adding  `**"prepare": "husky install”**` script needs to be added to the root level package.json. TIL: npm [`prepare` ](https://docs.npmjs.com/cli/v10/using-npm/scripts) scripts are magical and tap into the npm life cycle and will run implicitly when you run `npm install` locally.  

3.  “npm run nuke-and-install”  is a new command in our root package.json that will remove your node modules and install fresh dependecies in your project.  It’s a good idea to run this after merging major changes since we have people running multiple versions of node with different settings on this project. If you are noticing problems with your package-lock.json, you should delete these before running this command.  I’ll be curious to know if the folks running windows can run this.



Fixes #226 #224

## Type of change





## How Can this be tested? Testing Plan to review this PR



1. Checkout branch, npm install, go to Signin or Signup page. and click the google button to sign in.  You should be routed to the InputForm and see a  data object logged to the console. 

2. & 3. run `npm nuke-and-install` in the root of the project this should install fresh dependencies as well as prepare husky.  Make a small change and try to commit. You should see a lot of feedback in the terminal showing tsc checks, linting and prettier. 




## Checklist:




## Additional Information:
